android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1260,11 +1261,17 @@ private void checkForYandexSE() {
         }
     }
 
+    public void showNotificationRationale() {
+        BraveNotificationPermissionRationaleDialog notificationRationaleDialog =
+                BraveNotificationPermissionRationaleDialog.newInstance();
+        notificationRationaleDialog.show(getSupportFragmentManager(),
+                BraveNotificationPermissionRationaleDialog.NOTIFICATION_RATIONALE_DIALOG_TAG);
+    }
+
     private void showNotificationWarningDialog() {
         BraveNotificationWarningDialog notificationWarningDialog =
                 BraveNotificationWarningDialog.newInstance(
                         BraveNotificationWarningDialog.FROM_LAUNCHED_BRAVE_ACTIVITY);
-        notificationWarningDialog.setCancelable(false);
```

Comment:
what was the reason to remove it ?

Reply:
Removed this line, because back key was not working.
If we want to force user to click on close( X mark) then we can have this line.
I feel user this dialog can close if click outside / press on back key.

Reply:
my suggestion would be to disable dismiss on outside click. as we are showing close button on the UI

Reply:
reverted this.
Added back that line
notificationWarningDialog.setCancelable(false);